src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,54 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.service.producer.BulldozerProducer;
+import core.mate.academy.service.producer.ExcavatorProducer;
+import core.mate.academy.service.producer.MachineProducer;
+import core.mate.academy.service.producer.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
```

Comment:
The class implementation should not begin with an empty line. Please remove the unnecessary empty lines at the start of the class definition.